data/mods/gen7/random-teams.ts

```diff
@@ -29,568 +30,666 @@ const ZeroAttackHPIVs: {[k: string]: SparseStatsTable} = {
+			if (movePool.includes('spikes')) this.fastPop(movePool, movePool.indexOf('spikes'));
+			if (moves.size + movePool.length <= this.maxMoveCount) return;
+		}
+
+		// Develop additional move lists
+		const pivotingMoves = ['partingshot', 'uturn', 'voltswitch'];
+		const badWithSetup = ['defog', 'dragontail', 'haze', 'healbell', 'nuzzle', 'pursuit', 'rapidspin', 'toxic'];
+		const statusMoves = this.dex.moves.all()
+			.filter(move => move.category === 'Status')
+			.map(move => move.id);
+
+		// These moves don't mesh well with other aspects of the set
+		this.incompatibleMoves(moves, movePool, statusMoves, ['healingwish', 'memento', 'switcheroo', 'trick']);
+		this.incompatibleMoves(moves, movePool, Setup, pivotingMoves);
+		this.incompatibleMoves(moves, movePool, Setup, Hazards);
+		this.incompatibleMoves(moves, movePool, Setup, badWithSetup);
+		this.incompatibleMoves(moves, movePool, PhysicalSetup, PhysicalSetup);
+		this.incompatibleMoves(moves, movePool, SpeedSetup, ['quickattack', 'suckerpunch']);
+		this.incompatibleMoves(moves, movePool, 'defog', Hazards);
+		this.incompatibleMoves(moves, movePool, ['fakeout', 'uturn'], ['switcheroo', 'trick']);
+		this.incompatibleMoves(moves, movePool, 'substitute', pivotingMoves);
+		this.incompatibleMoves(moves, movePool, 'leechseed', 'dragontail');
+		this.incompatibleMoves(moves, movePool, 'rest', 'substitute');
+		this.incompatibleMoves(moves, movePool, PhysicalSetup, 'dracometeor');
+		this.incompatibleMoves(moves, movePool, SpecialSetup, 'knockoff');
+
+		if (!types.includes('Normal')) {
+			this.incompatibleMoves(moves, movePool, Setup, 'Explosion');
+		}
+
+		// These attacks are redundant with each other
+		this.incompatibleMoves(moves, movePool, 'psychic', 'psyshock');
+		this.incompatibleMoves(moves, movePool, 'scald', ['hydropump', 'originpulse', 'waterpulse']);
+		this.incompatibleMoves(moves, movePool, 'return', ['bodyslam', 'doubleedge']);
+		this.incompatibleMoves(moves, movePool, ['fierydance', 'firelash', 'lavaplume'], ['fireblast', 'magmastorm']);
+		this.incompatibleMoves(moves, movePool, ['flamethrower', 'flareblitz'], ['fireblast', 'overheat']);
+		this.incompatibleMoves(moves, movePool, 'hornleech', 'woodhammer');
+		this.incompatibleMoves(moves, movePool, ['gigadrain', 'leafstorm'], ['leafstorm', 'petaldance', 'powerwhip']);
+		this.incompatibleMoves(moves, movePool, 'wildcharge', 'thunderbolt');
+		this.incompatibleMoves(moves, movePool, 'gunkshot', 'poisonjab');
+		this.incompatibleMoves(moves, movePool, ['drainpunch', 'focusblast'], ['closecombat', 'highjumpkick', 'superpower']);
+		this.incompatibleMoves(moves, movePool, 'stoneedge', 'headsmash');
+		this.incompatibleMoves(moves, movePool, 'dracometeor', 'dragonpulse');
+		this.incompatibleMoves(moves, movePool, 'dragonclaw', 'outrage');
+		this.incompatibleMoves(moves, movePool, 'knockoff', ['darkestlariat', 'darkpulse', 'foulplay']);
+		if (!types.includes('Dark') && preferredType !== 'Dark') {
```

Comment:
I think these are more suited to be in the body of the method, especially since many of the incompatibilities are for specific Pokemon rather than all Pokemon in general. If anything, I think it would be more confusing to have even more long objects/arrays at the top of the file, I'd prefer for top-level constants to be only for the most general-purpose things. From experience in gen 9, having them in the body does not make them any harder to modify.

Reply:
What do you mean by "for specific Pokemon"? The ones I grabbed for the review comment don't have any indication that they're targeted at certain species. Anyway, the benefit is you wouldn't have repeat `this.incompatibleMoves(moves, movePool,` all the time, since you could just iterate over the incompatibility array.

Reply:
That makes sense, I went ahead and put the general incompatibilities in its own array. Since it's only relevant in this single context, I think it still belongs in the method and not at the top of the file.

Reply:
OK; having the data in the method seems a bit counterintuitive to me but if it's beneficial to randbats coders I suppose it's fine.

I still don't understand why the incompatibilies ostensibly for particular species aren't in the array; it might be faster to only run those if you're generating for a particular species, but you run `incompatibleMoves` on them regardless of species! Is 
```ts
// Lunatone
this.incompatibleMoves(moves, movePool, 'moonlight', 'rockpolish');
```
somehow better than a commented entry in the array:
```ts
const incompatiblePairs = [
    ...,
    // For Lunatone
    ['moonlight', 'rockpolish'],
    ...,
];
```

Reply:
Sure, I've moved the applicable hardcodes into the array